### PR TITLE
refactor: new sinsp logger implementation

### PR DESCRIPTION
**Pull request by therealbobo**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

/kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

This removes the `g_logger` extern variable in favour of a new a more maintainable api. The logger can now the accessed with `sinsp::get_logger()`.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Hi! Thanks for this PR!
While i like the idea, i have some concerns regarding build time, since everywhere sinsp logger is used must now include `sinsp.h` :/ 

Also, i'd just call the method `::logger` instead of `::get_logger`, since there is no setter.

File: userspace/libsinsp/sinsp.h [hunk 1]
```diff
@@ -1323,3 +1328,79 @@ VISIBILITY_PRIVATE
 };
 
 /*@}*/
+
+#define SINSP_LOG_(severity, fmt, ...)                                         \
```
Comment (Andreagit97) [inline userspace/libsinsp/sinsp.h:1332]: are these macros used somewhere?

Comment (therealbobo) [reply userspace/libsinsp/sinsp.h:1332]: They are used somewhere in the libs. I didn't remove them just to be conservative: maybe some libs consumer uses them. But, if we all agree, I can remove them :)

Comment (Andreagit97) [reply userspace/libsinsp/sinsp.h:1332]: uhm got it! I would be on board to remove them, WDYT @falcosecurity/libs-maintainers? In the worst scenario, we could re-introduce them if some adopter wants to rely on them

**Comment by gnosek** [comment]
Yeah, I don't like adding even more responsibilities to the sinsp class :/

If naming is the only issue, can we have e.g. `sinsp_logger` be a function so we can `sinsp_logger()->log(...)`?

File: userspace/libsinsp/sinsp.h (hunk 1 above)
Comment (gnosek) [reply userspace/libsinsp/sinsp.h:1332]: Move them to a separate header? We can include it from sinsp.h or not (or remove it after a while and if nobody complains, remove the file entirely?)

**Comment by therealbobo** [comment]
Didn't thought to do it that way! I'll update the pr :)

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by incertum — APPROVED** [review]
/approve

**Pull request merged**